src/monodroid/jni/monodroid-glue.c

```diff
@@ -1502,19 +1502,19 @@ gc_prepare_for_java_collection (JNIEnv *env, int num_sccs, MonoGCBridgeSCC **scc
 		 * Solution: Make all objects within the SCC directly or indirectly reference each other
 		 */
 		if (scc->num_objs > 1) {
-			MonoGCBridgeSCC *first = scc->objs [0];
+			MonoGCBridgeSCC *first = (MonoGCBridgeSCC*)scc->objs [0];
```

Comment:
With the most recent commit, `MonoObject` is now `void`, so this cast shouldn't be necessary anymore.